### PR TITLE
Convert ARM IP drivers to DT_INST

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: drivers/serial/uart_pl011.c [hunk 1]
```diff
@@ -519,26 +529,26 @@ static void pl011_irq_config_func_1(struct device *dev)
...
-	irq_enable(DT_PL011_PORT1_IRQ_TX);
+	irq_enable(DT_INST_1_ARM_PL011_IRQ_TX);
 
-	IRQ_CONNECT(DT_PL011_PORT1_IRQ_RX,
-		    DT_PL011_PORT1_IRQ_PRI,
+	IRQ_CONNECT(DT_INST_1_ARM_PL011_IRQ_RX,
+		    DT_INST_1_ARM_PL011_IRQ_RX_PRIORITY,
 		    pl011_isr,
-		    DEVICE_GET(pl011_port_1),
+		    DEVICE_GET(pl011_port_0),
```
Comment (MaureenHelm) [inline drivers/serial/uart_pl011.c]: Don't think this should be changed

File: drivers/serial/uart_pl011.c [hunk 2]
```diff
@@ -519,26 +529,26 @@ static void pl011_irq_config_func_1(struct device *dev)
...
-	irq_enable(DT_PL011_PORT1_IRQ_RX);
+	irq_enable(DT_INST_1_ARM_PL011_IRQ_RX);
 
-	IRQ_CONNECT(DT_PL011_PORT1_IRQ_RXTIM,
-		    DT_PL011_PORT1_IRQ_PRI,
+	IRQ_CONNECT(DT_INST_1_ARM_PL011_IRQ_RXTIM,
+		    DT_INST_1_ARM_PL011_IRQ_RXTIM_PRIORITY,
 		    pl011_isr,
-		    DEVICE_GET(pl011_port_1),
+		    DEVICE_GET(pl011_port_0),
```
Comment (MaureenHelm) [inline drivers/serial/uart_pl011.c]: Don't think this should be changed

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Looks like you accidentally changed port_1 to port_0 in a few spots.

File: drivers/serial/uart_pl011.c (hunk 1 above)
Comment (galak) [reply drivers/serial/uart_pl011.c]: thanks, good catch.

**Review by MaureenHelm — APPROVED** [review]

**Comment by carlocaione** [comment]
As shippable is reporting also the MMU code should be changed..

**Review by ioannisg — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

File: drivers/serial/uart_pl011.c (hunk 1 above)
Comment (mnkp) [reply drivers/serial/uart_pl011.c]: The PR got merged but the issue seems to be still unresolved. Do we need a follow up patch?

Comment (galak) [reply drivers/serial/uart_pl011.c]: Yes, sorry about that.  Pushed PR #23488 for that.

**Pull request merged**